vega_sim/scenario/common/agents.py

```diff
@@ -1274,12 +1274,17 @@ def _liq_provis(self, state: VegaState) -> LiquidityProvision:
         buy_specs = [["PEGGED_REFERENCE_BEST_BID", 5, 1]]
         sell_specs = [["PEGGED_REFERENCE_BEST_ASK", 5, 1]]
 
+        if (self.curr_asks is not None) and (self.curr_bids is not None):
+            est_mid_price = (self.curr_bids[0].price + self.curr_asks[0].price) * 0.5
+        elif state is not None:
```

Comment:
This needs to either just be an `else` or have a third option

Reply:
I see from matching up the if conditions that we probably can't hit the condition, but I don't like leaving a variable uninitialised (as it would be in the `else` here) when it's used later. In this case, we can probably just set `est_mid_price = None` in the else block (later code could handle the case where it is None, but can't handle the case where it doesn't exist at all).

Reply:
Ah fair point, I agree. Have made that change 👍